src/main/java/strategy/Strategy.java

```diff
@@ -0,0 +1,49 @@
+package strategy;
+
+import java.util.List;
+import service.FruitService;
+import service.impl.FruitServiceImpl;
+
+public class Strategy {
+    private FruitService fruitService = new FruitServiceImpl();
+
+    public String getReport() {
+        return fruitService.getFruitReport();
+    }
+
+    private void balance(String name, int amount) {
+        fruitService.put(name, amount);
+    }
+
+    private void supply(String name, int amount) {
+        int add = fruitService.get(name);
+        fruitService.put(name, add + amount);
+    }
+
+    private void purchase(String name, int amount) {
+        int bought = fruitService.get(name);
+        fruitService.put(name, bought - amount);
+    }
```

Comment:
You should create separate classes for each operation
Interface OperationHandler, and classes like PurchaseOperationHandler, SupplyOperationHandler etc